crates/ra_syntax/src/parsing.rs

```diff
@@ -11,7 +11,7 @@ use crate::{
     syntax_node::GreenNode,
 };
 
-pub use self::lexer::{tokenize, Token};
+pub use self::lexer::{tokenize, next_token, Token};
```

Comment:
Instead of exposing `next_token` as is, let's make a separate `pub fn classify_literal(&str)`, which uses `next_token` internally, but checks that

* this is indeed a literal and not, say, an identifier
* that all input is consume (that is, that the length of the first token is equal to the length of the whole str)